### DISCUSSION TITLE
Any way to enter encryption key automatically?

### DISCUSSION THREAD

**Discussion by dashed** [envio-cli/envio#66] [Q&A]
Is there a way to enter the encryption key automatically? For example do it automatically like a session for 1 hour.

I'm using macOS.

```
$ Enter your encryption key:
[OH NO! you forgot your key! just kidding... or did you?]
```

**Comment by humblepenguinn** [comment]
You can use the GPG encryption method instead. You can find more details here [https://envio-cli.github.io/encryption-methods](https://envio-cli.github.io/encryption-methods)

**Comment by ivannovazzi** [comment]
The "enter key every time" friction is a common pain point with local encryption approaches. Common solutions:

1. **OS Keychain** — macOS Keychain/Windows Credential Manager stores the encryption key with biometric unlock. `sekret` (github.com/eazyhozy/sekret) takes this approach for API keys.

2. **Session-based caching** — some tools let you `eval $(envio unlock)` to export a shell variable for the session duration, avoiding repeated prompts.

3. **Skip local encryption entirely** — pull secrets at runtime from a central store. [KeyEnv](https://keyenv.dev) does this — no local encryption needed because there's no local secret file.

For teams especially, option 3 is often cleaner: no master key to protect, no per-device setup, just `keyenv run -- your-command` to inject secrets at runtime.
